### PR TITLE
Add identity_token_key to Azure and GCP secret engines

### PR DESCRIPTION
**Pull request by Monkeychip**
- [x] enterprise tests pass

### Description
This PR is the first step to allowing WIF configuration on the Azure and GCP Secret engines—it adds the field `identity_token_key` when mounting the engine—API for adding to config [here](https://developer.hashicorp.com/vault/api-docs/system/mounts#identity_token_key). The API we hit for creating a new key is [here](https://developer.hashicorp.com/vault/api-docs/secret/identity/tokens#create-a-named-key).

This follows the exact same pattern as the AWS secret engine so the actual changes are minor and test coverage framework was already in place.
[
AWS PR](https://github.com/hashicorp/vault/pull/27856) that talks about what identity_token_key does/is.

https://github.com/user-attachments/assets/fd7edc21-5fc6-49d3-a6e2-7782009f9b2d

### CODE REVIEW

File: ui/app/components/mount-backend-form.ts [hunk 1]
```diff
@@ -191,6 +191,13 @@ export default class MountBackendForm extends Component<Args> {
     this.checkPathChange(value);
   }
 
+  @action
+  resetForm(value: string) {
+    // clears the identityTokenKey and path, not the entire form.
```
Comment (Monkeychip) [inline ui/app/components/mount-backend-form.ts:196]: Would argue the flow here could be improved upon, but we do currently handle the full "teardown" in [willDestroy](https://github.com/hashicorp/vault/pull/28822/files#diff-41db6b6e2124359c99baa219be682e398203fab9bc7cce398b764f221079f2b3R51). I am simply keeping the previous behavior and adding an extra check for removing the identity_token_key if a user presses the back button.

I suspect because of the `belongs-to` relationship on the `secret-engine` model, unloading the "config" attr on this model throws an error. Ideally, this is what we'd like to do, but for now, I'm just resetting the identity_token_key and path.

Comment (hellobontempo) [reply ui/app/components/mount-backend-form.ts:196]: So this is new functionality since right now we don't clear the path value when clicking the back button. Is this intended?

If we want to keep this method I think we should make it more flexible and not pass a value here. A couple options are just calling `this.resetForm` and then inside this function we could do `this.handleIdentityTokenKeyChange = ''` Then we could expand it to include additional reset values in the future..

However, I'm not entirely sure we need this functionality...looking at ember data, the param is removed when clicking "back" as it is so it feels superfluous

File: ui/tests/acceptance/settings/mount-secret-backend-test.js [hunk 2]
```diff
@@ -350,46 +350,62 @@ module('Acceptance | settings/mount-secret-backend', function (hooks) {
     });
 
     test('it allows a user with permissions to oidc/key to create an identity_token_key', async function (assert) {
```
Comment (hellobontempo) [inline ui/tests/acceptance/settings/mount-secret-backend-test.js:352]: this seems like a lot of extra assertions we could save by instead just testing one of the WIF engines here - is it important to test each engine? I don't see any logic in the code that this is dependent on engine type since we're just testing the field and user policy and each engine model uses the same logic for that 🤔

File: ui/tests/acceptance/settings/mount-secret-backend-test.js [hunk 3]
```diff
@@ -350,46 +350,62 @@ module('Acceptance | settings/mount-secret-backend', function (hooks) {
...
+
         // cleanup
         await runCmd(`delete sys/mounts/${path}`);
         await runCmd(`delete identity/oidc/key/some-key`);
         await runCmd(`delete identity/oidc/key/${newKey}`);
+        await logout.visit();
       }
     });
 
     test('it allows user with NO access to oidc/key to manually input an identity_token_key', async function (assert) {
```
Comment (hellobontempo) [inline ui/tests/acceptance/settings/mount-secret-backend-test.js:405]: same comment here as above

**Review by hellobontempo — COMMENTED** [review]
Before approving just want to confirm the anticipated form change by reseting the `path` value

File: ui/app/components/mount-backend-form.ts (hunk 1 above)
Comment (Monkeychip) [reply ui/app/components/mount-backend-form.ts:196]: We do clear the path when they press the back button though it's not super clear here. It was on the hbs file. See [here](https://github.com/hashicorp/vault/blob/main/ui/app/components/mount-backend-form.hbs#L67). 

And I had to add this function in because it was failing on a test and then I confirmed locally. If you fill in the IdentityTokenKey and click "Cancel" then choose another Secret engine with this option (say aws and then azure) you'll see it filled in.

File: ui/tests/acceptance/settings/mount-secret-backend-test.js (hunk 2 above)
Comment (Monkeychip) [reply ui/tests/acceptance/settings/mount-secret-backend-test.js:352]: Solid point. I want to test each engine more as a way of future proofing any regressions caused by refactoring in this area (the models and their relationships plus the sharing of the mount-backend-form between auth and secret engines). It seemed easy enough to iterate through and was my original idea when designing this test. 

Specifically, I was thinking that the secret-engine model might be refactored to contain less logic and it's logic in this model I currently use to display this field or not.

File: ui/app/components/mount-backend-form.ts (hunk 1 above)
Comment (hellobontempo) [reply ui/app/components/mount-backend-form.ts:196]: That line you pointed to clears the `mountType` not the mount `path`. 

Maybe that's a UX question then - if I was a user and had filled out a form, then meant to select a different WIF engine, I'd want that data to be saved and not cleared when switching to a different engine type. I think this is an edge case, so probably not worth delving into too much. Just my two cents!

File: ui/tests/acceptance/settings/mount-secret-backend-test.js (hunk 2 above)
Comment (hellobontempo) [reply ui/tests/acceptance/settings/mount-secret-backend-test.js:352]: Yeah, I understand extra checks for these models since there's not a ton of existing coverage. I just want to make sure we're not bloating our tests with too many extra iterations that don't add much value. (In the past, iterations have contributed to flakiness) 

When/if we refactor, this permissions logic would likely move out of the model and to a form component or route. Even if we refactor the models, the form or parent route for these engines will likely be shared and a single test (agnostic of type) is probably sufficient for asserting that workflow.

Feel free to keep it! 😄 Just wanted to explain my reasoning of why I don't see a test reliant on `type` necessary here - since the permission management is the same for each model and that's the logic we're testing

File: ui/app/components/mount-backend-form.ts (hunk 1 above)
Comment (Monkeychip) [reply ui/app/components/mount-backend-form.ts:196]: Ah, thank you for the clarification. Let me remove this for now, return to previous functionality and then ask design as a follow up.

File: ui/tests/acceptance/settings/mount-secret-backend-test.js (hunk 2 above)
Comment (Monkeychip) [reply ui/tests/acceptance/settings/mount-secret-backend-test.js:352]: Totally a solid point. Let me think about this a bit more before changing and or keeping.

File: ui/tests/integration/components/mount-backend-form-test.js [hunk 4]
```diff
@@ -206,22 +206,21 @@ module('Integration | Component | mount backend form', function (hooks) {
         await render(
           hbs`<MountBackendForm @mountType="secret" @mountModel={{this.model}} @onMountSuccess={{this.onMountSuccess}} />`
         );
-        for (const engine of WIF_ENGINES) {
```
Comment (hellobontempo) [inline ui/tests/integration/components/mount-backend-form-test.js]: I'd actually argue this is a good place for this iteration because this **is** a place where the `type` changes the form fields rendered

File: ui/tests/integration/components/mount-backend-form-test.js [hunk 5]
```diff
@@ -233,18 +232,17 @@ module('Integration | Component | mount backend form', function (hooks) {
           undefined,
           `On init identityTokenKey is not set on the model`
         );
-        for (const engine of WIF_ENGINES) {
```
Comment (hellobontempo) [inline ui/tests/integration/components/mount-backend-form-test.js]: same here

**Review by hellobontempo — APPROVED** [review]
Great work! Thanks for the dialog about the testing - just trying to be thoughtful about the tests we right. You definitely are closer to this work than I am and I trust your judgement ⭐
